usefulUtilities/statusIndicator/hifiScript/statusIndicator.js

```diff
@@ -0,0 +1,210 @@
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+(function () {
+    var DEBUG = true;
+
+    // #region STATUS OVERLAY
+
+    // Create clickable status overlay on desktop
+    // In upper right hand corner of screen
+    // Displays Available or Busy and toggles after click
+    var desktopOverlay,
+        rectangleOverlay,
+        // desktop overlayOptions
+        OVERLAY_AVAILABLE_TEXT = "     Available",
+        OVERLAY_NOT_AVAILABLE_TEXT = "     Busy",
+        OVERLAY_WINDOW_RIGHT_PADDING = 10,
+        OVERLAY_WIDTH = 10,
+        OVERLAY_FONT_SIZE = 20.0,
+        OVERLAY_WIDTH = 200,
+        // rectangleOverlay options
+        OVERLAY_RECTANGLE_LEFT_PADDING = 10,
+        OVERLAY_RECTANGLE_TOP_PADDING = 12,
+        OVERLAY_RECTANGLE_WIDTH_HEIGHT = 20,
+        AVAILABLE_COLOR = { red: 0, green: 144, blue: 54 },
+        NOT_AVAILABLE_COLOR = { red: 255, green: 0, blue: 26 };
+    function drawStatusOverlays() {
+        var windowWidth = Window.innerWidth;
+
+        desktopOverlay = Overlays.addOverlay("text", {
+            text: isAvailable ? OVERLAY_AVAILABLE_TEXT : OVERLAY_NOT_AVAILABLE_TEXT,
+            width: OVERLAY_WIDTH,
+            height: OVERLAY_FONT_SIZE * 2.2,
+            x: windowWidth - OVERLAY_WIDTH - OVERLAY_WINDOW_RIGHT_PADDING,
+            y: OVERLAY_FONT_SIZE,
+            lineHeight: OVERLAY_FONT_SIZE,
+            margin: OVERLAY_FONT_SIZE / 2,
+            font: { size: OVERLAY_FONT_SIZE },
+            color: { red: 255, green: 255, blue: 255 },
+            backgroundColor: { red: 17, green: 17, blue: 17 },
+            alpha: 1.0,
+            backgroundAlpha: 0.7,
+            visible: true
+        });
+
+        rectangleOverlay = Overlays.addOverlay("rectangle", {
+            width: OVERLAY_RECTANGLE_WIDTH_HEIGHT,
+            height: OVERLAY_RECTANGLE_WIDTH_HEIGHT,
+            x: windowWidth - OVERLAY_WIDTH + OVERLAY_RECTANGLE_LEFT_PADDING - OVERLAY_WINDOW_RIGHT_PADDING,
+            y: OVERLAY_FONT_SIZE + OVERLAY_RECTANGLE_TOP_PADDING,
+            color: isAvailable ? AVAILABLE_COLOR : NOT_AVAILABLE_COLOR,
+            alpha: 1.0,
+            visible: true
+        });
+    }
+
+
+    // Delete clickable status overlay on desktop
+    function deleteStatusOverlays() {
+        if (rectangleOverlay) {
+            Overlays.deleteOverlay(rectangleOverlay);
+        }
+        if (desktopOverlay) {
+            Overlays.deleteOverlay(desktopOverlay);
+        }
+    }
+
+    // #endregion STATUS OVERLAY
+
+
+    // #region HEARTBEAT
+
+    // Send heartbeat with updates to database
+    // When this stops, the database will set status to Offline
+    var HEARTBEAT_INTERVAL_MS = 5000,
+        heartbeatInterval;
+    function setupHeartbeat() {
+        if (!heartbeatInterval) {
+            heartbeatInterval = Script.setInterval(function() {
```

Comment:
Rather than an interval, this should be a timeout. The timeout should be stopped (if one exists) then started inside `sendStatusUpdate()` after getting the response from the server.

Reply:
(lemme know if you have questions about why)